File: app/src/main/java/com/hfad/mymovies/api/ApiService.java
package com.hfad.mymovies.api;

import com.hfad.mymovies.data.models.ExampleResponse;
import com.hfad.mymovies.data.models.ExampleReview;
import com.hfad.mymovies.data.models.Movie;
import com.hfad.mymovies.data.models.MovieDiscover;

import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface ApiService {
    @GET("discover/movie?api_key=756d1951529fd2f2bc0df1846271d514&include_adult=false&include_video=false}")
    Observable<ExampleResponse> getExample(@Query("language") String language,@Query("page") int page, @Query("sort_by") String sort, @Query("vote_count.gte") int count);

    @GET("movie/{movie_id}?api_key=756d1951529fd2f2bc0df1846271d514")
    Observable<Movie> getExampleMovie(@Path("movie_id") int movieId, @Query("language") String language);

    @GET("movie/{movie_id}/credits?api_key=756d1951529fd2f2bc0df1846271d514")
    Observable<ExampleResponse> getCast(@Path("movie_id") int movieId, @Query("language") String language);

    @GET("search/movie?api_key=756d1951529fd2f2bc0df1846271d514")
    Observable<ExampleResponse> getSearchMovies(@Query("language") String language, @Query("query") String name);

    @GET("movie/{movie_id}/recommendations?api_key=756d1951529fd2f2bc0df1846271d514&include_adult=false&include_video=false}")
    Observable<ExampleResponse> getRecommended(@Path("movie_id") int movieId, @Query("language") String language,@Query("page") int page);

    @GET("movie/upcoming?api_key=756d1951529fd2f2bc0df1846271d514")
    Observable<ExampleResponse> getUpcoming(@Query("language") String language, @Query("page") int page);


    @GET("movie/{movie_id}/reviews?api_key=756d1951529fd2f2bc0df1846271d514")
    Observable<ExampleReview> getReviews(@Path("movie_id") int movieId, @Query("language") String language);
}

File: app/src/main/java/com/hfad/mymovies/api/ApiService.kt
package com.hfad.mymovies.api

import com.hfad.mymovies.data.models.ExampleResponse
import com.hfad.mymovies.data.models.ExampleReview
import com.hfad.mymovies.data.models.Movie
import io.reactivex.Observable
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

const val API_KEY = "api_key=756d1951529fd2f2bc0df1846271d514"
interface ApiService {
    @GET("discover/movie?$API_KEY&include_adult=false&include_video=false}")
    fun getExample(
            @Query("language") language: String?,
            @Query("page") page: Int,
            @Query("sort_by") sort: String?,
            @Query("vote_count.gte") count: Int
    ): Observable<ExampleResponse>

    @GET("movie/{movie_id}?$API_KEY")
    fun getExampleMovie(
            @Path("movie_id") movieId: Int,
            @Query("language") language: String?
    ): Observable<Movie>

    @GET("movie/{movie_id}/credits?$API_KEY")
    fun getCast(
            @Path("movie_id") movieId: Int,
            @Query("language") language: String?
    ): Observable<ExampleResponse>

    @GET("search/movie?$API_KEY")
    fun getSearchMovies(
            @Query("language") language: String?,
            @Query("query") name: String?
    ): Observable<ExampleResponse>

    @GET("movie/{movie_id}/recommendations?$API_KEY&include_adult=false&include_video=false}")
    fun getRecommended(
            @Path("movie_id") movieId: Int,
            @Query("language") language: String?,
            @Query("page") page: Int
    ): Observable<ExampleResponse>

    @GET("movie/upcoming?$API_KEY")
    fun getUpcoming(
            @Query("language") language: String?,
            @Query("page") page: Int
    ): Observable<ExampleResponse>

    @GET("movie/{movie_id}/reviews?$API_KEY")
    fun getReviews(
            @Path("movie_id") movieId: Int,
            @Query("language") language: String?
    ): Observable<ExampleReview>
}